### PR TITLE
Adding support for bhosphorus22k dataset

### PR DESCRIPTION
**Pull request by ManideepLadi**

### CODE REVIEW

File: openhands/datasets/isolated/bhosporus22k.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        
+        for i in range(df.shape[0]):
+            file_name=self.root_dir+"/"+format((df.iloc[i][1]),"04")+"/"+(df.iloc[i][4])+"_"+format((df.iloc[i][-1]),"03")+file_format
+            signer_id=df.iloc[i][4].split("_")[-1]
+            signer_id=int(signer_id)
+            gloss=(df.iloc[i][2])
+            gloss_cat = self.label_encoder.transform([gloss.strip(' \n\t')])[0]
+            if (
+                ((signer_id) !=4 and "train" in self.splits)
+                or (signer_id == 4 and "val" in self.splits)
```
Comment (GokulNC) [inline openhands/datasets/isolated/bhosporus22k.py:40]: @ManideepLadi What about test set? What signer_id for that? 
Does this mean the accuracies you reported were on valset and not the testset?

File: openhands/datasets/isolated/bhosporus22k.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    Turkis Isolated Sign language dataset(Bhosporus22k) from the paper:
+    Link to paper: https://arxiv.org/pdf/2004.01283.pdf
+    """
+    def read_glosses(self):
+        df = pd.read_csv(self.class_mappings_file_path, delimiter=",")
+        print(self.class_mappings_file_path)
+        all_glosses = [df.iloc[i][2] for i in range(len(df))]
+        unique_glosses=set(all_glosses)
+        self.glosses = list(unique_glosses)
+        #print(self.glosses)
```
Comment (GokulNC) [inline openhands/datasets/isolated/bhosporus22k.py:18]: Please remove lines like this after debugging/development is done.

File: openhands/datasets/isolated/bhosporus22k.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    def read_original_dataset(self):
+        """
+        Dataset includes 3200 videos where 10 non-expert subjects executed 5 repetitions of 64 different types of signs.
+
+        For train-set, we use signers 1-8.
+        Val-set & Test-set: Signer-9 & Signer-10
```
Comment (GokulNC) [inline openhands/datasets/isolated/bhosporus22k.py:25]: 1. Can you please add a comment here about what you actually found out about the `signer_id`s?
2. What signer_id did the SOTA paper use? (which you said had 93% accuracy)
3. Can you please write an email to the Bhosporus asking if the signer_id you found out (for both val & test) is the correct thing?

File: openhands/datasets/isolated/bhosporus22k.py (hunk 1 above)
Comment (ManideepLadi) [reply openhands/datasets/isolated/bhosporus22k.py:40]: According to the paper, signer-4 as test set and remaining signers as training set. So we used signer-4 as validation set and remaining as train set. So here validation set is the test set.

File: openhands/datasets/isolated/bhosporus22k.py (hunk 2 above)
Comment (ManideepLadi) [reply openhands/datasets/isolated/bhosporus22k.py:18]: Will remove those lines and will update the file.

File: openhands/datasets/isolated/bhosporus22k.py (hunk 1 above)
Comment (GokulNC) [reply openhands/datasets/isolated/bhosporus22k.py:40]: Got it, thanks! As a convention, for all datasets which do not have "val" set, we are calling it as "test" set explicitly.
Can you please rename "val" as "test"? (in this line as well as the config files)

Comment (ManideepLadi) [reply openhands/datasets/isolated/bhosporus22k.py:40]: Sure will do that Gokul.

File: openhands/datasets/isolated/bhosporus22k.py (hunk 3 above)
Comment (GokulNC) [reply openhands/datasets/isolated/bhosporus22k.py:25]: 4. Why does the description say val-set & test-set if there's no val-set? Please edit it if this is incorrect.

**Review by GokulNC — APPROVED** [review]

**Comment by GokulNC** [comment]
BTW, the commits does not seem to have your email ID properly. Please ensure from next time your commits have your name. :)
